### PR TITLE
Longhaul Tests: Eliminate unused twin test pipeline configuration in favor of hardcoded config

### PR DESCRIPTION
**Pull request by and-rewsmith**
My stance on the state of our yamls is that we have over-engineered configuration on most test pipelines which increases the complexity of making changes, leads to nasty bugs, and results in confusing behavior. Therefore I think removing unused configuration will scale better.

If we ever need to change these values, we can create a PR branch and manually change the default values in the `trcE2ETest.sh` script. This doesn't require rebuilding images.

In this PR I am doing the following:
- Removing the unused configuration for twin tests from the pipeline
- In order to follow the pattern where we have configuration that is defaulted in the `trcE2ETest.sh` script, I added `TwinUpdateFailureThreshold` for consistency.

### CODE REVIEW

File: builds/e2e/templates/nested-longhaul-deploy.yaml [hunk 1]
```diff
@@ -23,8 +23,6 @@ parameters:
   logAnalytics.sharedKey: ''
   logAnalytics.logType: ''
   twinTester.twinUpdateSize: ''
-  twinTester.twinUpdateFailureThreshold: ''
-  twinTester.edgeHubRestartFailureTolerance: ''
```
Comment (and-rewsmith) [inline builds/e2e/templates/nested-longhaul-deploy.yaml:27]: We were never even using these values.

**Comment by and-rewsmith** [comment]
Builds running to verify change. Longhaul and nested longhaul look good.

File: builds/e2e/templates/nested-longhaul-deploy.yaml (hunk 1 above)
Comment (yophilav) [reply builds/e2e/templates/nested-longhaul-deploy.yaml:27]: Was it you who made these param changes a while ago?

**Review by yophilav — APPROVED** [review]

File: builds/e2e/templates/nested-longhaul-deploy.yaml (hunk 1 above)
Comment (and-rewsmith) [reply builds/e2e/templates/nested-longhaul-deploy.yaml:27]: No. These param headers were copied from the original longhaul deployment without being integrated in:

https://github.com/Azure/iotedge/blame/39d324910d76bc6ebe62a90e9d4ddc77f8b53be3/builds/e2e/templates/nested-longhaul-deploy.yaml#L26

https://github.com/Azure/iotedge/pull/4333/files#diff-3e30d323ba6fc94aa3edf4e4cf11b37ffafbc81bfb4c02c8fcddb0759ab183d7R27

**Pull request merged**